Runar/Runar/Controllers/Generator flow/GenerationPopUpViewController.swift

```diff
@@ -142,76 +177,102 @@ public class GenerationPopUpViewController: UIViewController {
-            escapeButton.heightAnchor.constraint(equalToConstant: 48)
-        ])
+        escapeButton.addTarget(self, action: #selector(self.closeOnTap), for: .touchUpInside)
+        escapeButton.snp.makeConstraints { make in
+            make.top.equalTo(containerView.snp.top).offset(CGFloat.escapeButtonTopAnchor)
+            make.trailing.equalTo(containerView.snp.trailing).inset(CGFloat.escapeButtonTrailingAnchor)
+            make.width.height.equalTo(CGFloat.escapeButtonWidth)
+        }
         
         containerView.addSubview(imageView)
+        imageView.snp.makeConstraints { make in
+            make.top.equalTo(containerView.snp.top).offset(CGFloat.imageViewTopAnchor)
+            make.leading.equalTo(containerView.snp.leading).offset(CGFloat.imageViewLeadingAnchor)
+            make.width.equalTo(CGFloat.imageViewWidth)
+            make.height.equalTo(CGFloat.imageViewHeight)
+        }
+        
+        containerView.addSubview(headerLabel)
+        headerLabel.snp.makeConstraints { make in
+            make.leading.equalTo(imageView.snp.trailing).offset(CGFloat.headerLabelLeadingAnchot)
+            make.height.equalTo(CGFloat.headerLabelHeight)
+            make.centerY.equalTo(imageView.snp.centerY)
+        }
         
-        imageView.translatesAutoresizingMaskIntoConstraints = false
-        NSLayoutConstraint.activate([
-            imageView.topAnchor.constraint(equalTo: containerView.topAnchor, constant: 18),
-            imageView.leftAnchor.constraint(equalTo: containerView.leftAnchor, constant: 20),
-            imageView.heightAnchor.constraint(equalToConstant: 80),
-            imageView.widthAnchor.constraint(equalToConstant: 80)
-        ])
-        
-        containerView.addSubview(header)
-        
-        header.translatesAutoresizingMaskIntoConstraints = false
-        NSLayoutConstraint.activate([
-            header.topAnchor.constraint(equalTo: containerView.topAnchor, constant: 40),
-            header.leftAnchor.constraint(equalTo: imageView.rightAnchor, constant: 26),
-            header.heightAnchor.constraint(equalToConstant: 42)
-        ])
-        
-        containerView.addSubview(desc)
-        
-        desc.translatesAutoresizingMaskIntoConstraints = false
-        NSLayoutConstraint.activate([
-            desc.topAnchor.constraint(equalTo: containerView.topAnchor, constant: 100),
-            desc.leftAnchor.constraint(equalTo: containerView.leftAnchor, constant: 30),
-            desc.widthAnchor.constraint(equalToConstant: 300),
-            desc.heightAnchor.constraint(equalToConstant: 130)
-        ])
+        containerView.addSubview(descriptionTextView)
+        descriptionTextView.snp.makeConstraints { make in
+            make.top.equalTo(imageView.snp.bottom).offset(CGFloat.descriptionTextViewTopAnchor)
+            make.leading.trailing.equalTo(containerView).inset(CGFloat.descriptionTextViewLeadingAnchor)
+            make.width.equalTo(CGFloat.descriptionTextViewWidth)
```

Comment:
Conflict constraint
Try to guess why?

Reply:
I think you mean that I can't use width constraint with leading and trailing.